app/services/ContractorCYAService.scala

```diff
@@ -40,28 +44,38 @@ class ContractorCYAService @Inject()(cisSessionService: CISSessionService,
         cisConnector.submit(user.nino, taxYear, submission)(hc.withExtraHeaders(headers = "mtditid" -> user.mtditid)).flatMap {
           case Left(error) => Future.successful(Left(HttpParserError(error.status)))
           case Right(_) =>
-            val handleAuditingFuture = handleAuditing(taxYear, employerRef, user, cisUserData)
```

Comment:
I think it is better if you split the auditing and the NRS calls. This will simplify the method.